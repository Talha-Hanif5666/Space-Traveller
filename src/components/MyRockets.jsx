import React from 'react';
import PropTypes from 'prop-types';
import { ListGroup } from 'react-bootstrap';

function MyRockets({ myRockets }) {
  let content = '';
  if (myRockets.length !== 0) {
    content = (
      <ListGroup>
        {myRockets.map((rocket) => <ListGroup.Item key={rocket.id} className="fw-bold">{rocket.name}</ListGroup.Item>)}
      </ListGroup>
    );
  } else {
    content = <p role="status">No rockets yet</p>;
  }
  return (
    <>
      <h2>Rockets booked</h2>
      {content}
    </>
  );
}

export default MyRockets;

MyRockets.propTypes = {
  myRockets: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    desc: PropTypes.string.isRequired,
    booked: PropTypes.bool.isRequired,
  })).isRequired,
};
