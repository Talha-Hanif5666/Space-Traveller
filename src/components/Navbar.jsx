import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { NavLink } from 'react-router-dom';
import logo from '../assets/planet.png';

function BasicExample() {
  return (
    <Navbar bg="light" expand="lg">
      <Container>
        <Navbar.Brand href="#home">
          <img
            src={logo}
            width="50"
            height="50"
            className="d-inline-block align-top"
            alt="React Bootstrap logo"
          />
          <span style={{ fontWeight: 'bold', fontSize: '30px', marginLeft: '10px' }}>Space Traveler&apos;s Hub</span>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="navbar-nav" />
        <Navbar.Collapse id="navbar-nav">
          <Nav className="ms-auto" as="ul" variant="pills">
            <Nav.Item as="li" className="mx-2">
              <NavLink to="/rockets" style={({ isActive }) => ({ color: isActive ? 'blue' : 'black' })}>Rockets</NavLink>
            </Nav.Item>
            <Nav.Item as="li" className="mx-2">
              <NavLink to="/missions" style={({ isActive }) => ({ color: isActive ? 'blue' : 'black' })}>Missions</NavLink>
            </Nav.Item>
            <Nav.Item as="li" className="mx-2">
              <NavLink to="/my-profile" style={({ isActive }) => ({ color: isActive ? 'blue' : 'black' })}>My Profile</NavLink>
            </Nav.Item>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export default BasicExample;
