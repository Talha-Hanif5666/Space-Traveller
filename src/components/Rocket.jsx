import React from 'react';
import PropTypes from 'prop-types';
import { Badge, Button } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { booking } from '../redux/rockets/rocketsSlice';

function Rocket({ rocket }) {
  const dispatch = useDispatch();
  return (
    <li className="list-group-item">
      <div className="card mb-3">
        <div className="row g-0">
          <div className="col-md-4">
            <img src={rocket.img} className="img-fluid rounded-start" alt="..." />
          </div>
          <div className="col-md-8">
            <div className="card-body">
              <h5 className="card-title">{rocket.name}</h5>
              <p className="card-text">
                <Badge role="status" bg={rocket.booked ? 'secondary' : 'primary'} pill>{rocket.booked ? 'Reserved' : 'Available'}</Badge>
                {rocket.desc}
              </p>
              <Button role="button" onClick={() => dispatch(booking(rocket.id))} variant={rocket.booked ? 'secondary' : 'primary'}>{rocket.booked ? 'Cancel reservation' : 'Reserve rocket'}</Button>
            </div>
          </div>
        </div>
      </div>
    </li>
  );
}

export default Rocket;

Rocket.propTypes = {
  rocket: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    desc: PropTypes.string.isRequired,
    booked: PropTypes.bool.isRequired,
  })).isRequired,
};
