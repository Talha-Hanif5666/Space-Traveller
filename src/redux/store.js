/* eslint-disable import/no-extraneous-dependencies */
import thunk from 'redux-thunk';
import { legacy_createStore as createStore, combineReducers, applyMiddleware } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missions from './missions/missionsSlice';

const reducers = combineReducers({
  rockets: rocketsReducer,
  missionsReducer: missions,
});

const store = createStore(reducers, applyMiddleware(thunk));

export default store;
