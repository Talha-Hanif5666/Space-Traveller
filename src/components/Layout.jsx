import React from 'react';
import PropTypes from 'prop-types';
import { Outlet } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Navbar from './Navbar';

function Layout({ error, loading }) {
  return (
    <>
      <Navbar error={error} loading={loading} />
      <Container>
        <Outlet />
      </Container>
    </>
  );
}

export default Layout;

Layout.propTypes = {
  error: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
};
