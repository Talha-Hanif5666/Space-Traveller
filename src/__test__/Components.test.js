import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Layout from '../components/Layout';
import MyRockets from '../components/MyRockets';
import Navbar from '../components/Navbar';
import Rocket from '../components/Rocket';

describe('Layout tests', () => {
  test('to render Layout with an error: ', () => {
    const mockedStatus = [
      {
        error: 'An error message',
        loading: false,
      },
    ];
    const layout = render(
      <Provider store={store}>
        <BrowserRouter>
          <Layout status={mockedStatus} />
        </BrowserRouter>
      </Provider>,
    );
    expect(layout).toMatchSnapshot();
  });
  test('to render Layout when loading and no error: ', () => {
    const mockedStatus = [
      {
        error: '',
        loading: true,
      },
    ];
    const layout = render(
      <Provider store={store}>
        <BrowserRouter>
          <Layout status={mockedStatus} />
        </BrowserRouter>
      </Provider>,
    );
    expect(layout).toMatchSnapshot();
  });
});

describe('MyRockets tests', () => {
  test('to render MyRockets with all rockets booked: ', () => {
    const mockedRockets = [
      {
        id: 'rocket1',
        name: 'Rocket 1',
        desc: 'A rocket description',
        img: 'rocket1.jpg',
        booked: true,
      },
      {
        id: 'rocket2',
        name: 'Rocket 2',
        desc: 'A rocket description',
        img: 'rocket2.jpg',
        booked: true,
      },
    ];
    const myRockets = render(
      <Provider store={store}>
        <BrowserRouter>
          <MyRockets myRockets={mockedRockets} />
        </BrowserRouter>
      </Provider>,
    );
    expect(myRockets).toMatchSnapshot();
  });

  it('renders Calculator and test if key pressed updates value of calculator display', () => {
    const mockedRockets = [];
    render(<MyRockets myRockets={mockedRockets} />);
    expect(screen.getByRole('status')).toHaveTextContent('No rockets yet');
  });
});

describe('Navbar tests', () => {
  test('to render errors: ', () => {
    const mockedError = 'An error message';
    const mockedLoading = false;
    const navbar = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar error={mockedError} loading={mockedLoading} />
        </BrowserRouter>
      </Provider>,
    );
    expect(navbar.getByRole('log')).toHaveTextContent('An error message');
  });

  test('to render loading: ', () => {
    const mockedError = '';
    const mockedLoading = true;
    const navbar = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar error={mockedError} loading={mockedLoading} />
        </BrowserRouter>
      </Provider>,
    );
    expect(navbar.getByRole('status')).toHaveTextContent('Loading...');
  });
});

describe('Rocket element tests', () => {
  test('to render rocket element: ', () => {
    const mockedRocket = {
      id: 'rocket1',
      name: 'Rocket 1',
      desc: 'A rocket description',
      img: 'rocket1.jpg',
      booked: false,
    };

    const rocket = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rocket rocket={mockedRocket} />
        </BrowserRouter>
      </Provider>,
    );
    expect(rocket).toMatchSnapshot();
  });

  test('to render button as available for booking: ', () => {
    const mockedRocket = {
      id: 'rocket1',
      name: 'Rocket 1',
      desc: 'A rocket description',
      img: 'rocket1.jpg',
      booked: false,
    };

    const rocket = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rocket rocket={mockedRocket} />
        </BrowserRouter>
      </Provider>,
    );
    expect(rocket.getByRole('button')).toHaveTextContent('Reserve rocket');
    expect(rocket.getByRole('status')).toHaveTextContent('Available');
  });

  test('to render button as available for booking: ', () => {
    const mockedRocket = {
      id: 'rocket1',
      name: 'Rocket 1',
      desc: 'A rocket description',
      img: 'rocket1.jpg',
      booked: true,
    };

    const rocket = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rocket rocket={mockedRocket} />
        </BrowserRouter>
      </Provider>,
    );
    expect(rocket.getByRole('button')).toHaveTextContent('Cancel reservation');
    expect(rocket.getByRole('status')).toHaveTextContent('Reserved');
  });
});
