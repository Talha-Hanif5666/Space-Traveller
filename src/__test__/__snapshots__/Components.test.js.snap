// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests to render Layout when loading and no error:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="#home"
          >
            <img
              alt="React Bootstrap logo"
              class="d-inline-block align-top"
              height="50"
              src="planet.png"
              width="50"
            />
            <span
              style="font-weight: bold; font-size: 30px; margin-left: 10px;"
            >
              Space Traveler's Hub
            </span>
          </a>
          <button
            aria-controls="navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="navbar-nav"
          >
            <ul
              class="ms-auto navbar-nav nav-pills"
            >
              <li
                class="mx-2 nav-item"
              >
                <a
                  class=""
                  href="/rockets"
                  style="color: black;"
                >
                  Rockets
                </a>
              </li>
              <li
                class="mx-2 nav-item"
              >
                <a
                  class=""
                  href="/missions"
                  style="color: black;"
                >
                  Missions
                </a>
              </li>
              <li
                class="mx-2 nav-item"
              >
                <a
                  class=""
                  href="/my-profile"
                  style="color: black;"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="container"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="#home"
        >
          <img
            alt="React Bootstrap logo"
            class="d-inline-block align-top"
            height="50"
            src="planet.png"
            width="50"
          />
          <span
            style="font-weight: bold; font-size: 30px; margin-left: 10px;"
          >
            Space Traveler's Hub
          </span>
        </a>
        <button
          aria-controls="navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="navbar-nav"
        >
          <ul
            class="ms-auto navbar-nav nav-pills"
          >
            <li
              class="mx-2 nav-item"
            >
              <a
                class=""
                href="/rockets"
                style="color: black;"
              >
                Rockets
              </a>
            </li>
            <li
              class="mx-2 nav-item"
            >
              <a
                class=""
                href="/missions"
                style="color: black;"
              >
                Missions
              </a>
            </li>
            <li
              class="mx-2 nav-item"
            >
              <a
                class=""
                href="/my-profile"
                style="color: black;"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="container"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Layout tests to render Layout with an error:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="#home"
          >
            <img
              alt="React Bootstrap logo"
              class="d-inline-block align-top"
              height="50"
              src="planet.png"
              width="50"
            />
            <span
              style="font-weight: bold; font-size: 30px; margin-left: 10px;"
            >
              Space Traveler's Hub
            </span>
          </a>
          <button
            aria-controls="navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="navbar-nav"
          >
            <ul
              class="ms-auto navbar-nav nav-pills"
            >
              <li
                class="mx-2 nav-item"
              >
                <a
                  class=""
                  href="/rockets"
                  style="color: black;"
                >
                  Rockets
                </a>
              </li>
              <li
                class="mx-2 nav-item"
              >
                <a
                  class=""
                  href="/missions"
                  style="color: black;"
                >
                  Missions
                </a>
              </li>
              <li
                class="mx-2 nav-item"
              >
                <a
                  class=""
                  href="/my-profile"
                  style="color: black;"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="container"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="#home"
        >
          <img
            alt="React Bootstrap logo"
            class="d-inline-block align-top"
            height="50"
            src="planet.png"
            width="50"
          />
          <span
            style="font-weight: bold; font-size: 30px; margin-left: 10px;"
          >
            Space Traveler's Hub
          </span>
        </a>
        <button
          aria-controls="navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="navbar-nav"
        >
          <ul
            class="ms-auto navbar-nav nav-pills"
          >
            <li
              class="mx-2 nav-item"
            >
              <a
                class=""
                href="/rockets"
                style="color: black;"
              >
                Rockets
              </a>
            </li>
            <li
              class="mx-2 nav-item"
            >
              <a
                class=""
                href="/missions"
                style="color: black;"
              >
                Missions
              </a>
            </li>
            <li
              class="mx-2 nav-item"
            >
              <a
                class=""
                href="/my-profile"
                style="color: black;"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="container"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyRockets tests to render MyRockets with all rockets booked:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        Rockets booked
      </h2>
      <div
        class="list-group"
      >
        <div
          class="fw-bold list-group-item"
        >
          Rocket 1
        </div>
        <div
          class="fw-bold list-group-item"
        >
          Rocket 2
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h2>
      Rockets booked
    </h2>
    <div
      class="list-group"
    >
      <div
        class="fw-bold list-group-item"
      >
        Rocket 1
      </div>
      <div
        class="fw-bold list-group-item"
      >
        Rocket 2
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket element tests to render rocket element:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="list-group-item"
      >
        <div
          class="card mb-3"
        >
          <div
            class="row g-0"
          >
            <div
              class="col-md-4"
            >
              <img
                alt="..."
                class="img-fluid rounded-start"
                src="rocket1.jpg"
              />
            </div>
            <div
              class="col-md-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Rocket 1
                </h5>
                <p
                  class="card-text"
                >
                  <span
                    class="badge rounded-pill bg-primary"
                    role="status"
                  >
                    Available
                  </span>
                  A rocket description
                </p>
                <button
                  class="btn btn-primary"
                  role="button"
                  type="button"
                >
                  Reserve rocket
                </button>
              </div>
            </div>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="list-group-item"
    >
      <div
        class="card mb-3"
      >
        <div
          class="row g-0"
        >
          <div
            class="col-md-4"
          >
            <img
              alt="..."
              class="img-fluid rounded-start"
              src="rocket1.jpg"
            />
          </div>
          <div
            class="col-md-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Rocket 1
              </h5>
              <p
                class="card-text"
              >
                <span
                  class="badge rounded-pill bg-primary"
                  role="status"
                >
                  Available
                </span>
                A rocket description
              </p>
              <button
                class="btn btn-primary"
                role="button"
                type="button"
              >
                Reserve rocket
              </button>
            </div>
          </div>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
